x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.js

```diff
@@ -27,57 +23,97 @@ export class HeadlessChromiumDriverFactory {
 
   type = 'chromium';
 
-  create({ viewport, browserTimezone }) {
-    return Rx.Observable.create(async observer => {
-      const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
-      const chromiumArgs = args({
+  async create({ viewport, browserTimezone }) {
+    const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
+    const chromiumArgs = args({
+      userDataDir,
+      viewport,
+      verboseLogging: this.logger.isVerbose,
+      disableSandbox: this.browserConfig.disableSandbox,
+      proxyConfig: this.browserConfig.proxy,
+    });
+
+    let chromium;
+    let page;
+
+    try {
+      chromium = await puppeteer.launch({
         userDataDir,
-        viewport,
-        verboseLogging: this.logger.isVerbose,
-        disableSandbox: this.browserConfig.disableSandbox,
-        proxyConfig: this.browserConfig.proxy,
+        executablePath: this.binaryPath,
+        ignoreHTTPSErrors: true,
+        args: chromiumArgs,
+        env: {
+          TZ: browserTimezone
+        },
       });
 
-      let chromium;
-      let page;
-      try {
-        chromium = await puppeteer.launch({
-          userDataDir,
-          executablePath: this.binaryPath,
-          ignoreHTTPSErrors: true,
-          args: chromiumArgs,
-          env: {
-            TZ: browserTimezone
-          },
-        });
-
-        page = await chromium.newPage();
-      } catch (err) {
-        observer.error(new Error(`Caught error spawning Chromium`));
-        return;
-      }
 
-      safeChildProcess({
-        async kill() {
-          await chromium.close();
+      page = await chromium.newPage();
+    } catch (err) {
+      throw new Error(`Caught error spawning Chromium` + err);
+      return;
+    }
+
+    page.on('console', line => {
+      if (line._type === 'error') {
+        if (line._text.match(/\[\d+\/\d+.\d+:\w+:CONSOLE\(\d+\)\]/)) {
+          if (line._text.includes('error while loading shared libraries: libnss3.so')) {
+            throw new Error(`You must install nss for Reporting to work`);}
+          if (line._text.includes('Check failed: InitDefaultFont(). Could not find the default font')) {
+            throw new Error('You must install freetype and ttf-font for Reporting to work');}
```

Comment:
These error messages are currently really focused around *NIX systems. Not sure if you wouldn't get the same error message on a windows system? I think we should try to produce the same errors on Windows and check if chromium will report the same messages. If they are different I think we can perfectly fine use those *NIX specific messages, if not we should come up with better messages. Maybe we should also link to the reporting setup documentation here.